### PR TITLE
Exclude Kotlin libraries from the plugin zip

### PR DESCRIPTION
**Pull request by PawelLipski**
Not confident enough to ship it in 2.0.1 yet

Zip size change: 9.7 MB -> 5.6 MB... not that it matters too much 👅 

Zip contents diff:

image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -70,9 +70,11 @@ jobs:
           command: ./gradlew test
       # Unfortunately, wildcards for test result paths aren't supported by CircleCI yet.
       - store_test_results:
-          path: branchLayoutImpl/build/test-results/test/
+          path: branchLayout/impl/build/test-results/test/
```
Comment (PawelLipski) [inline .circleci/config.yml:73]: Unrelated fix, one last (?) thing left from the module reorganization

File: build.gradle [hunk 2]
```diff
@@ -18,10 +18,10 @@ plugins {
...
   id 'org.jetbrains.grammarkit' version '2021.2.2' apply false
   id 'org.jetbrains.intellij' version '1.6.0' apply false
-  // Minor Kotlin version is effectively enforced by the IntelliJ version we're building against,
-  // check https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library
-  // every time `intellijVersions.buildTarget` is bumped.
-  id 'org.jetbrains.kotlin.jvm' version '1.6.20' apply false
+  // As per https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library:
+  //   If a plugin supports multiple platform versions, it must (...) target the lowest bundled stdlib version
+  // which is 1.4.0 in our case (as this is the Kotlin version bundled into IntelliJ 2020.3).
+  id 'org.jetbrains.kotlin.jvm' version '1.4.0' apply false
```
Comment (PawelLipski) [inline build.gradle:24]: Downgrade
not-stonks

File: frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt [hunk 3]
```diff
@@ -25,7 +25,7 @@ class OverrideForkPointDialog(
   init {
     title =
         getString("action.GitMachete.BaseOverrideForkPointAction.dialog.override-fork-point.title")
-    setOKButtonMnemonic('O'.code)
+    setOKButtonMnemonic('O'.toInt())
```
Comment (PawelLipski) [inline frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt:28]: Side effect of the downgrade, `.code` showed up only in 1.5 ☹️

File: gradle.properties [hunk 4]
```diff
@@ -2,3 +2,4 @@ org.gradle.caching=true
 org.gradle.configureOnDemand=true
 org.gradle.jvmargs=-Xmx6G
 org.gradle.parallel=true
+kotlin.stdlib.default.dependency=false
```
Comment (PawelLipski) [inline gradle.properties:5]: The crucial change: tell Gradle to no longer bundle `kotlin-stdlib` into plugin zip.

Kotlin libraries are provided by IntelliJ anyway! In the same manner, we don't need to bundle e.g. IntelliJ API libraries into our plugin... they're provided OOTB when plugin runs in the IDE

File: scripts/verify-artifact-contents [hunk 5]
```diff
@@ -35,14 +35,14 @@ done
...
-expected_libraries=(org.eclipse.jgit kotlin-reflect kotlin-stdlib reflections slf4j-lambda-core vavr vavr-match)
+expected_libraries=(org.eclipse.jgit reflections slf4j-lambda-core vavr vavr-match)
 for lib in ${expected_libraries[@]}; do
   grep -qEx "${lib//./\\.}-[0-9.]+.*\.jar" <<< "$jars_in_plugin_zip" || fail "$lib jar is expected but missing from $plugin_zip"
 done
 
 # We don't want either slf4j-api or any SLF4J implementation jars in the final plugin artifact
-# as this would lead to a linker error (for api) or an SLF4J error (for impl) in the runtime.
-forbidden_library_prefixes=(checker ide-probe idea junit-driver lombok remote-robot scala slf4j)
+# as this might lead to a linker error (for api) or an SLF4J error (for impl) in the runtime.
```
Comment (PawelLipski) [inline scripts/verify-artifact-contents:44]: There's so much nuance here that I don't want to elaborate 😅

File: config/checker/git4idea.astub [hunk 6]
```diff
@@ -1,6 +1,6 @@
+import org.checkerframework.checker.nullness.qual.Nullable;
 import org.checkerframework.checker.tainting.qual.Tainted;
 import org.checkerframework.checker.tainting.qual.Untainted;
-import org.jetbrains.annotations.Nullable;
```
Comment (PawelLipski) [inline config/checker/git4idea.astub:3]: Meh, didn't compile when Kotlin was downgraded, for some weird reason

**Review by azdrojowa123 — APPROVED** [review]

File: build.gradle [hunk 7]
```diff
@@ -18,10 +18,10 @@ plugins {
   id 'org.checkerframework' version '0.6.11' apply false
   id 'org.jetbrains.grammarkit' version '2021.2.2' apply false
   id 'org.jetbrains.intellij' version '1.6.0' apply false
-  // Minor Kotlin version is effectively enforced by the IntelliJ version we're building against,
-  // check https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library
-  // every time `intellijVersions.buildTarget` is bumped.
-  id 'org.jetbrains.kotlin.jvm' version '1.6.20' apply false
+  // As per https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library:
+  //   If a plugin supports multiple platform versions, it must (...) target the lowest bundled stdlib version
```
Comment (mkondratek) [inline build.gradle:22]: `(...)`?

File: build.gradle [hunk 8]
```diff
@@ -424,16 +424,16 @@ allprojects {
...
-      }
 
-      def kotlinVersion = plugins.getPlugin(org.jetbrains.kotlin.gradle.plugin.KotlinPluginWrapper.class).kotlinPluginVersion
-      dependencies {
-        // kotlin-reflect is NOT added by Gradle Kotlin plugin 1.4/1.5/1.6 automatically,
-        // even though it turns out to be needed in the runtime (e.g. in `DeleteBranchOnSlideOutSuggestionDialog`).
-        // Note that kotlin-reflect pulls in kotlin-stdlib transitively, even if `kotlin.stdlib.default.dependency=false` is set in gradle.properties
-        // (see https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library)
-        implementation group: 'org.jetbrains.kotlin', name: 'kotlin-reflect', version: kotlinVersion
+        // Supress the warnings about different version of Kotlin used for compilation
```
Comment (mkondratek) [inline build.gradle:428]: 👍

**Review by mkondratek — APPROVED** [review]

File: version.gradle [hunk 9]
```diff
@@ -1,3 +1,3 @@
 ext {
-  PROSPECTIVE_RELEASE_VERSION = '2.0.1'
+  PROSPECTIVE_RELEASE_VERSION = '2.0.2'
```
Comment (PawelLipski) [inline version.gradle:2]: Whoops, I was supposed NOT to merge it :trollface:
